mxcube3/routes/SampleCentring.py

```diff
@@ -264,8 +294,16 @@ def update_shapes():
         # Shape does not have any refs, create a new Centered position
         if not refs:
             x, y = shape_data["screen_coord"]
-            mpos = mxcube.diffractometer.get_centred_point_from_coord(x, y)
-            shape = mxcube.shapes.add_shape_from_mpos([mpos], (x, y), t)
+            mpos = mxcube.diffractometer.get_centred_point_from_coord(x, y, return_by_names=True)
+            pos.append(mpos)
+            # MD3 needs the center of the grid instead of the top left corner
```

Comment:
mmm.. I having trouble for moving down in the hierarchy, neither collect_hwobj or diff_hwobj know anything about the shape selected or shape's details/params. I do not want to start creating custom hwobj for the shapes...

What about each grid has two positions (top-left and center) as I did here before, but when creating the acquisition parameters I select which cpos to use for the mesh:

```
mesh_center = mxcube.beamline['default_mesh_values'].getProperty('mesh_center', 'top-left')

if mesh_center == 'top-left':
    acq.acquisition_parameters.centred_position = grid.get_centred_positions()[0]
 else:
    acq.acquisition_parameters.centred_position = grid.get_centred_positions()[1]
```
Unless you have a better idea...